packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -218,8 +218,9 @@ export class CacheDataClient implements IDataClient {
       const now = new Date();
 
       if (now >= deadline) {
-        this.logger.error('Unable to connect to Momento: deadline exceeded.');
-        resolve();
+        const errorMessage = 'Unable to connect to Momento: deadline exceeded.';
+        this.logger.error(errorMessage);
+        reject(new Error(errorMessage));
```

Comment:
can we throw one of our sdk errors here instead of a generic error?

Reply:
yeah added!